### DISCUSSION TITLE
IntelliSense not working as expected with imported functions in JS project.

### DISCUSSION THREAD

**Discussion by sergetk** [typescript-language-server/typescript-language-server#902] [Q&A]
Hello.

I'am seen imported method reported as variable instead of function  after first completion in js file. 
Initial auto-complete/importing works as expected but all subsequent auto-complete attempts report function as variable.

I have created simple js file with named function declaration.
![function](https://github.com/typescript-language-server/typescript-language-server/assets/25514908/bff583a5-d9b2-4f3c-9aef-c5a91f94bf95)

I then auto-complete this function name in another file and it shows it correctly as a function inside auto-complete popup.
![first_auto_comple_func](https://github.com/typescript-language-server/typescript-language-server/assets/25514908/46f3d2f5-327a-4511-b90e-a13b4637e966)

Unfortunately all the subsequent attempts to auto-complete this function name will show it as a variable instead of a function,
This results in loss of function parameters list.
![second_auto_complete_var](https://github.com/typescript-language-server/typescript-language-server/assets/25514908/10cefc87-f480-499f-913a-af25ab4f1d16)

Observation, if I remove import statement at the top and try to auto-complete again, IntelliSense once again shows it as a function.

Tested in Emacs with LSP/Eglot and vscode. 

Is this a bug, or do I need to do some special configuration to get this to work correctly?

Thank you for you help.

Regards.

**Comment by rchl** [comment]
If it acts the same in VSCode then I'd suggest reporting in vscode (or Typescript) repo since that's where the logic lives.
